src/components/tokenInputPercentage/TokenInputPercentage.tsx

```diff
@@ -0,0 +1,86 @@
+import { BigNumber } from 'bignumber.js';
+import { TokenInputField } from 'components/tokenInputField/TokenInputField';
+import { useState, useMemo, useEffect } from 'react';
+import { Token } from 'services/observables/tokens';
+import { classNameGenerator } from 'utils/pureFunctions';
+
+interface TokenInputPercentageProps {
+  label: string;
+  token?: Token;
+  balance?: string;
+  amount: string;
+  setAmount: Function;
+}
+
+export const TokenInputPercentage = ({
+  token,
+  balance,
+  label,
+  amount,
+  setAmount,
+}: TokenInputPercentageProps) => {
+  const [amountUSD, setAmountUSD] = useState('');
```

Comment:
optional: lines 22-39 + the implementation of onClick (line 68-77) can go into a custom hook. It can also provide you with props object for TokenInputField - and then the component remains very simple